crone/src/SoftcutClient.cpp

```diff
@@ -138,7 +138,10 @@ void crone::SoftcutClient::handleCommand(Commands::CommandPacket *p) {
 	cut.setRecFlag(idx_0, value > 0.f);
 	break;
     case Commands::Id::SET_CUT_REC_ONCE_FLAG:
-	cut.setRecOnceFlag(idx_0, value > 0.f);
+	cut.setRecOnceFlag(idx_0-1, idx_1 > 0);
```

Comment:
few issues:
- needs `clamp(idx_0, NumVoices-1);`

- reconciling 1-based indices shouldn't happen here. (this module should not have to know that commands are coming from lua.) the only code module that should know such things is `weaver.c`, converting from 1-base immediately at the interface from lua to C, as here: https://github.com/monome/norns/blob/main/matron/src/weaver.c#L2302
(if you come across instances where this isn't happening then we should fix them..)
(oh btw, sending to supercollider engines is different. there, we have no way of knowing if a value is an index or lnot, so we don't touch it.)

- doesn't seem to me like we actually want/need a third argument. i would remove `_flag` from the API name and just have this command enable the `rec_once` behavior for the next cut, with the second argument to optionally initiate a cut... what do you think?

- even if the 3rd argument is used i would do `iff`, using a float for the flag value as with other flag-type softcut parameters.

Reply:
I'm in agreement with everything.

I do like the idea of removing the flag and going to just two arguments, like `rec_once(<voice>,<position>)` where `<position>` is optional. one thing though - what about a scenario where you accidentally run `rec_once` on a super huge loop and want to cancel it - where should the cancellation be available (if it should)? maybe one way is to have a `rec(<voice>,0)` also dual purpose to cancel any `rec_once` initiations?

Reply:
that did occur to me. but it seems like an unusual circumstance (maybe i'm wrong) and there are some options:
- agree it would be good for touching `rec_flag` to also cancel the `recOnce` state
- (but since one goal for the `rec_once` behavior is avoiding clicks, would also be nice to not require that...)
- could add a specific `rec_once_cancel` command
- as you say, can sequence a couple position changes
- `rec_level` is always an option

Reply:
or of course if it seems best to reuquire explicit flag state argument, i have no problem with that and woul djust change the flag value to  a float.